lib/matplotlib/tests/test_axes.py

```diff
@@ -4429,6 +4429,57 @@ def test_empty_shared_subplots():
     assert y1 >= 6
 
 
+def test_shared_with_aspect():
```

Comment:
Any reason not to turn this into 3 separate tests? This is going to stop running at the first failed assert, which seems like a bad thing when stuff breaks.

Reply:
Good point.  I split it.